apps/server/src/modules/authentication/testing/icurrentuser.factory.ts

```diff
@@ -0,0 +1,71 @@
+import { BaseFactory } from '@shared/testing';
+import { ICurrentUser } from '../interface';
+
+class CurrentUser implements ICurrentUser {
+	userId: string;
+
+	roles: string[];
+
+	schoolId: string;
+
+	accountId: string;
+
+	systemId: string;
+
+	isExternalUser: boolean;
+
+	constructor(data: ICurrentUser) {
+		this.userId = data.userId;
+		this.roles = data.roles;
+		this.schoolId = data.schoolId;
+		this.accountId = data.accountId;
+		this.systemId = data.systemId || '';
+		this.isExternalUser = data.isExternalUser;
+	}
+}
+
+export class ICurrentUserFactory extends BaseFactory<CurrentUser, ICurrentUser> {
+	public buildAdminICurrentUser(): ICurrentUser {

```

Comment:
I think you need to set the params here like in `ColumnBoardFactory` to include them in the normal creation process and not skip the creation with `sequence`. Otherwise these users are all the same.